controllers/vspherevm_controller.go

```diff
@@ -349,6 +354,28 @@ func (r vmReconciler) reconcileNormal(ctx *context.VMContext) (reconcile.Result,
 	return reconcile.Result{}, nil
 }
 
+func (r vmReconciler) isWaitingForStaticIPAllocation(ctx *context.VMContext) bool {
+	// Requeue if the two DHCP flags are set to false
+	// and no static IP is set in the IPAddrs.
+	waitForIP := false
+	devices := ctx.VSphereVM.Spec.Network.Devices
+
+	for _, dev := range devices {
+		// If DHCP is set for any of the devices,
+		// assume DHCP as the default allocation type.
+		if dev.DHCP4 || dev.DHCP6 {
+			return false
+		} else {
+			if len(dev.Gateway4) <= 0 || len(dev.IPAddrs) <= 0 {
```

Comment:
is Gateway6 check needed?

Reply:
Ack. The 2 gateways are required when the respective DHCP flags are false.

Reply:
At this point we don't know if IPAddrs has ipv4 or ipv6. So we cant determine which specific gateway(4/6) needs to be available. May be the gateway check is not required here.